api/v1/api.go

```diff
@@ -27,38 +36,105 @@ func NewApiV1() *apiV1 {
 // @type apiKey
 // @in header
 // @name Authorization
-func (s *apiV1) RegisterRoutes(e *echo.Echo) {
+func (s *apiV1) RegisterRoutes(e *echo.Echo, config config.DeltaConfig) {
```

Comment:
A few tweaks here - basically, can we clean up so it looks similar to the way we set up the routes in `Delta-dm`? 
i.e here
https://github.com/application-research/delta-dm/blob/main/api/router.go#L47

- remove the `apiV1` struct altogether, don't think it's needed 
- create an `echo` `apiGroup := e.Group("/api/v1")` and put all the routes under that 
- Move all the other middleware in here (ex `e.Use(middleware.RequestID())`)

Reply:
In the follow up prs I was going to move these functions into their right classes. This was more of skeleton set up for later so things could compile and the general layout would be there. I can move the middleware and api droups in this pr though